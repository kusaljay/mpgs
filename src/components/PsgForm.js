import React from 'react';
import {IoIosContact} from 'react-icons/io';

const PsgForm = ({inputChanged, submitGenerate, errors}) => {
  return (
    <div className="card content-container">
      <form onSubmit={submitGenerate}>
        <h2 className="psg-heading-card"><IoIosContact className="icon"/>Employee info</h2>
        {errors && <div className="alert alert-danger form-alerts">{errors}</div>}
        <div className="d-flex flex-wrap mr-n4">
          <div className="form-group empinfo-field">
            <label>First Name</label>
            <input type="text" className="form-control" name="firstName" onChange={inputChanged} />
          </div>
          <div className="form-group empinfo-field">
            <label>Last Name</label>
            <input type="text" className="form-control" name="lastName" onChange={inputChanged} />
          </div>
          <div className="form-group empinfo-field">
            <label>Annual Salary</label>
            <input type="text" className="form-control dollar" name="annualIncome" onChange={inputChanged} />
          </div>
          <div className="form-group empinfo-field">
            <label>Superannuation Rate</label>
            <input type="text" className="form-control" name="superPercent" onChange={inputChanged} />
          </div>
          <button type="submit" className="btn btn-success btn-post-login">Generate Payslip</button>
        </div>

      </form>
    </div>
  )
}

export default PsgForm;
