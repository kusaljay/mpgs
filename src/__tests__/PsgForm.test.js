import React from 'react';
import { shallow } from 'enzyme';
import PsgForm from '../components/PsgForm';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<PsgForm inputChanged submitGenerate errors />);
});

describe('PsgForm', () => {
  it('Should render form', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should validate when the button is clicked', () => {
    wrapper.find('button').simulate('click');
    wrapper.setProps({errors: true});
    expect(wrapper.find('.form-alerts')).toHaveLength(1);
  });
});