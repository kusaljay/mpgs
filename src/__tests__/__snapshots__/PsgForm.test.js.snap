// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsgForm Should render form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PsgForm
    errors={true}
    inputChanged={true}
    submitGenerate={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={true}
      >
        <h2
          className="psg-heading-card"
        >
          <IoIosContact
            className="icon"
          />
          Employee info
        </h2>
        <div
          className="alert alert-danger form-alerts"
        >
          true
        </div>
        <div
          className="d-flex flex-wrap mr-n4"
        >
          <div
            className="form-group empinfo-field"
          >
            <label>
              First Name
            </label>
            <input
              className="form-control"
              name="firstName"
              onChange={true}
              type="text"
            />
          </div>
          <div
            className="form-group empinfo-field"
          >
            <label>
              Last Name
            </label>
            <input
              className="form-control"
              name="lastName"
              onChange={true}
              type="text"
            />
          </div>
          <div
            className="form-group empinfo-field"
          >
            <label>
              Annual Salary
            </label>
            <span
              className="unit"
            >
              $
            </span>
            <input
              className="form-control empinfo-input"
              name="annualIncome"
              onChange={true}
              type="text"
            />
          </div>
          <div
            className="form-group empinfo-field"
          >
            <label>
              Superannuation Rate
            </label>
            <span
              className="unit"
            >
              %
            </span>
            <input
              className="form-control empinfo-input"
              name="superPercent"
              onChange={true}
              type="text"
            />
          </div>
          <button
            className="btn btn-success btn-post-login"
            type="submit"
          >
            Generate Payslip
          </button>
        </div>
      </form>,
      "className": "card content-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="psg-heading-card"
          >
            <IoIosContact
              className="icon"
            />
            Employee info
          </h2>,
          <div
            className="alert alert-danger form-alerts"
          >
            true
          </div>,
          <div
            className="d-flex flex-wrap mr-n4"
          >
            <div
              className="form-group empinfo-field"
            >
              <label>
                First Name
              </label>
              <input
                className="form-control"
                name="firstName"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Last Name
              </label>
              <input
                className="form-control"
                name="lastName"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Annual Salary
              </label>
              <span
                className="unit"
              >
                $
              </span>
              <input
                className="form-control empinfo-input"
                name="annualIncome"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Superannuation Rate
              </label>
              <span
                className="unit"
              >
                %
              </span>
              <input
                className="form-control empinfo-input"
                name="superPercent"
                onChange={true}
                type="text"
              />
            </div>
            <button
              className="btn btn-success btn-post-login"
              type="submit"
            >
              Generate Payslip
            </button>
          </div>,
        ],
        "onSubmit": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <IoIosContact
                className="icon"
              />,
              "Employee info",
            ],
            "className": "psg-heading-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "icon",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Employee info",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": true,
            "className": "alert alert-danger form-alerts",
          },
          "ref": null,
          "rendered": true,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="form-group empinfo-field"
              >
                <label>
                  First Name
                </label>
                <input
                  className="form-control"
                  name="firstName"
                  onChange={true}
                  type="text"
                />
              </div>,
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Last Name
                </label>
                <input
                  className="form-control"
                  name="lastName"
                  onChange={true}
                  type="text"
                />
              </div>,
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Annual Salary
                </label>
                <span
                  className="unit"
                >
                  $
                </span>
                <input
                  className="form-control empinfo-input"
                  name="annualIncome"
                  onChange={true}
                  type="text"
                />
              </div>,
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Superannuation Rate
                </label>
                <span
                  className="unit"
                >
                  %
                </span>
                <input
                  className="form-control empinfo-input"
                  name="superPercent"
                  onChange={true}
                  type="text"
                />
              </div>,
              <button
                className="btn btn-success btn-post-login"
                type="submit"
              >
                Generate Payslip
              </button>,
            ],
            "className": "d-flex flex-wrap mr-n4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    First Name
                  </label>,
                  <input
                    className="form-control"
                    name="firstName"
                    onChange={true}
                    type="text"
                  />,
                ],
                "className": "form-group empinfo-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "First Name",
                  },
                  "ref": null,
                  "rendered": "First Name",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "firstName",
                    "onChange": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Last Name
                  </label>,
                  <input
                    className="form-control"
                    name="lastName"
                    onChange={true}
                    type="text"
                  />,
                ],
                "className": "form-group empinfo-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Last Name",
                  },
                  "ref": null,
                  "rendered": "Last Name",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "lastName",
                    "onChange": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Annual Salary
                  </label>,
                  <span
                    className="unit"
                  >
                    $
                  </span>,
                  <input
                    className="form-control empinfo-input"
                    name="annualIncome"
                    onChange={true}
                    type="text"
                  />,
                ],
                "className": "form-group empinfo-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Annual Salary",
                  },
                  "ref": null,
                  "rendered": "Annual Salary",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$",
                    "className": "unit",
                  },
                  "ref": null,
                  "rendered": "$",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control empinfo-input",
                    "name": "annualIncome",
                    "onChange": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Superannuation Rate
                  </label>,
                  <span
                    className="unit"
                  >
                    %
                  </span>,
                  <input
                    className="form-control empinfo-input"
                    name="superPercent"
                    onChange={true}
                    type="text"
                  />,
                ],
                "className": "form-group empinfo-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Superannuation Rate",
                  },
                  "ref": null,
                  "rendered": "Superannuation Rate",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "%",
                    "className": "unit",
                  },
                  "ref": null,
                  "rendered": "%",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control empinfo-input",
                    "name": "superPercent",
                    "onChange": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Generate Payslip",
                "className": "btn btn-success btn-post-login",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Generate Payslip",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={true}
        >
          <h2
            className="psg-heading-card"
          >
            <IoIosContact
              className="icon"
            />
            Employee info
          </h2>
          <div
            className="alert alert-danger form-alerts"
          >
            true
          </div>
          <div
            className="d-flex flex-wrap mr-n4"
          >
            <div
              className="form-group empinfo-field"
            >
              <label>
                First Name
              </label>
              <input
                className="form-control"
                name="firstName"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Last Name
              </label>
              <input
                className="form-control"
                name="lastName"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Annual Salary
              </label>
              <span
                className="unit"
              >
                $
              </span>
              <input
                className="form-control empinfo-input"
                name="annualIncome"
                onChange={true}
                type="text"
              />
            </div>
            <div
              className="form-group empinfo-field"
            >
              <label>
                Superannuation Rate
              </label>
              <span
                className="unit"
              >
                %
              </span>
              <input
                className="form-control empinfo-input"
                name="superPercent"
                onChange={true}
                type="text"
              />
            </div>
            <button
              className="btn btn-success btn-post-login"
              type="submit"
            >
              Generate Payslip
            </button>
          </div>
        </form>,
        "className": "card content-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="psg-heading-card"
            >
              <IoIosContact
                className="icon"
              />
              Employee info
            </h2>,
            <div
              className="alert alert-danger form-alerts"
            >
              true
            </div>,
            <div
              className="d-flex flex-wrap mr-n4"
            >
              <div
                className="form-group empinfo-field"
              >
                <label>
                  First Name
                </label>
                <input
                  className="form-control"
                  name="firstName"
                  onChange={true}
                  type="text"
                />
              </div>
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Last Name
                </label>
                <input
                  className="form-control"
                  name="lastName"
                  onChange={true}
                  type="text"
                />
              </div>
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Annual Salary
                </label>
                <span
                  className="unit"
                >
                  $
                </span>
                <input
                  className="form-control empinfo-input"
                  name="annualIncome"
                  onChange={true}
                  type="text"
                />
              </div>
              <div
                className="form-group empinfo-field"
              >
                <label>
                  Superannuation Rate
                </label>
                <span
                  className="unit"
                >
                  %
                </span>
                <input
                  className="form-control empinfo-input"
                  name="superPercent"
                  onChange={true}
                  type="text"
                />
              </div>
              <button
                className="btn btn-success btn-post-login"
                type="submit"
              >
                Generate Payslip
              </button>
            </div>,
          ],
          "onSubmit": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <IoIosContact
                  className="icon"
                />,
                "Employee info",
              ],
              "className": "psg-heading-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "icon",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Employee info",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": true,
              "className": "alert alert-danger form-alerts",
            },
            "ref": null,
            "rendered": true,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="form-group empinfo-field"
                >
                  <label>
                    First Name
                  </label>
                  <input
                    className="form-control"
                    name="firstName"
                    onChange={true}
                    type="text"
                  />
                </div>,
                <div
                  className="form-group empinfo-field"
                >
                  <label>
                    Last Name
                  </label>
                  <input
                    className="form-control"
                    name="lastName"
                    onChange={true}
                    type="text"
                  />
                </div>,
                <div
                  className="form-group empinfo-field"
                >
                  <label>
                    Annual Salary
                  </label>
                  <span
                    className="unit"
                  >
                    $
                  </span>
                  <input
                    className="form-control empinfo-input"
                    name="annualIncome"
                    onChange={true}
                    type="text"
                  />
                </div>,
                <div
                  className="form-group empinfo-field"
                >
                  <label>
                    Superannuation Rate
                  </label>
                  <span
                    className="unit"
                  >
                    %
                  </span>
                  <input
                    className="form-control empinfo-input"
                    name="superPercent"
                    onChange={true}
                    type="text"
                  />
                </div>,
                <button
                  className="btn btn-success btn-post-login"
                  type="submit"
                >
                  Generate Payslip
                </button>,
              ],
              "className": "d-flex flex-wrap mr-n4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      First Name
                    </label>,
                    <input
                      className="form-control"
                      name="firstName"
                      onChange={true}
                      type="text"
                    />,
                  ],
                  "className": "form-group empinfo-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "First Name",
                    },
                    "ref": null,
                    "rendered": "First Name",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "firstName",
                      "onChange": true,
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Last Name
                    </label>,
                    <input
                      className="form-control"
                      name="lastName"
                      onChange={true}
                      type="text"
                    />,
                  ],
                  "className": "form-group empinfo-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Last Name",
                    },
                    "ref": null,
                    "rendered": "Last Name",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "lastName",
                      "onChange": true,
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Annual Salary
                    </label>,
                    <span
                      className="unit"
                    >
                      $
                    </span>,
                    <input
                      className="form-control empinfo-input"
                      name="annualIncome"
                      onChange={true}
                      type="text"
                    />,
                  ],
                  "className": "form-group empinfo-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Annual Salary",
                    },
                    "ref": null,
                    "rendered": "Annual Salary",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "$",
                      "className": "unit",
                    },
                    "ref": null,
                    "rendered": "$",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control empinfo-input",
                      "name": "annualIncome",
                      "onChange": true,
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Superannuation Rate
                    </label>,
                    <span
                      className="unit"
                    >
                      %
                    </span>,
                    <input
                      className="form-control empinfo-input"
                      name="superPercent"
                      onChange={true}
                      type="text"
                    />,
                  ],
                  "className": "form-group empinfo-field",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Superannuation Rate",
                    },
                    "ref": null,
                    "rendered": "Superannuation Rate",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "%",
                      "className": "unit",
                    },
                    "ref": null,
                    "rendered": "%",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control empinfo-input",
                      "name": "superPercent",
                      "onChange": true,
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Generate Payslip",
                  "className": "btn btn-success btn-post-login",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Generate Payslip",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
